Getting ready for v3.0.0

intervalia opened the pull request:
● Moved source files to lib folder.
● Added index.js.
● Cleaned up documentation.
● Now using rollup as a sub-object for require.

intervalia committed:
Getting ready for v3.0.0
● Moved source files to lib folder.
● Added index.js.
● Cleaned up documentation.
● Now using rollup as a sub-object for require.

GyozaGuy commented on package.json:
```diff
@@ -16,8 +16,9 @@
   },
   "dependencies": {
     "glob": "^7.1.2",
+    "rollup": "^0.60.7",
```
This seems like it might be important to have 🤔

intervalia replied:
Didn't need it before since I was only testing the output of my routines. But I needed to add tests that ran everything, including rollup. So now it is needed. here.

intervalia committed:
More cleanup
Added html-minifier to better minify the templates
Instead of just `{}` and `{rollup}` we now use `{components:{rollup{}}}` when exporting from index.js
Improved README

GyozaGuy commented on README.md:
```diff
@@ -165,13 +175,13 @@ If the value for `includePath` had been set to true then the output structure wo
 
 `srcFolder` is the only option that must be supplied. This specifies the folder or folders that are to be processed by the build tools.
```
Should `srcFolder` here be `srcFolders`? Or is `srcFolders` wrong where it is found?

intervalia replied:
fixed

GyozaGuy commented on README.md:
```diff
@@ -165,13 +175,13 @@ If the value for `includePath` had been set to true then the output structure wo
 `srcFolder` is the only option that must be supplied. This specifies the folder or folders that are to be processed by the build tools.
 
-If you have all of your components stored in child fodlers of the folder `./comps` then you would call `rollup.init({srcFolders:['./comps/*']})`.
+If you have all of your components stored in child fodlers of the folder `./comps` then you would call `init({srcFolders:['./comps/*']})`.
```
`fodlers` --> `folders`

intervalia replied:
fixed

GyozaGuy commented on README.md:
```diff
@@ -165,13 +175,13 @@ If the value for `includePath` had been set to true then the output structure wo
 
 Every folder directly under `./comps` would get processed. Any subfolder that was found to match the requirements of a component will be handed off to rollup for further processing.
 
 You can specify each folder independently within the array, And each entry in the array can either be a real path to a specific folder or a glob path to a set of folders.
```
The `And` after the `,` should not be capitalized.

intervalia replied:
fixed

GyozaGuy commented on test/testFolders/rollupFolders/MJSWithLocalesAndTemplates/one/one.mjs:
```diff
@@ -1,4 +1,6 @@
+import tempaltes from "./_compiled/templates.mjs";
```
`tempaltes` --> `templates`

intervalia replied:
fixed

GyozaGuy commented on test/testFolders/rollupFolders/MJSWithLocalesAndTemplates/one/one.mjs:
```diff
@@ -1,4 +1,6 @@
 const lang = locales();
 
 console.log(lang.BUTTON_OK);
+console.log(tempaltes.str('styles'));
```
`tempaltes` --> `templates`

intervalia replied:
fixed

intervalia committed:
Fixed coverage reports and added another test to get 100% coverage.

intervalia committed:
Spelling correction

intervalia committed:
More spelling corrections
